src/modules/AutoinstConfig.rb

```diff
@@ -522,6 +527,38 @@ def MainHelp
       main_help
     end
 
+    # Reports name of base product as obtained from AY profile
+    #
+    # There are several ways how can base product be defined in the profile
+    # 1) explicitly
+    # 2) impllicitly according software selection
+    #
+    # @return [Y2Product] a base product or nil
+    def selected_product
+      return @selected_product if @selected_product
+
+      # magic for selecting base product according to the profile
+      # check if base product is defined explicitly in the profile
+      software = Profile.current.fetch("software", {})
+      base_products = Y2Packager::Product.available_base_products
+
+      # try to find base product for installation according the profile
+      product_name = software.fetch("products", {})["product"]
+      return base_products.find { |p| p.short_name == product_name } if product_name
+
+      # try to find product according to patterns in profile
+      # searching for patterns like "sles-base-32bit"
+      products = base_products.select do |product|
+        software["patterns"].any? { |p| p =~ /#{product.name.downcase}-.*/ }
```

Comment:
Above we use `fetch` to access "products", yet here we assume "patterns" is always there. But it is not: https://github.com/yast/yast-autoinstallation/blob/90fdf50a1f6167b0dcaf20a5a589d693e2b382a1/src/autoyast-rnc/software.rnc#L12